plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/TrinoAvroSerDe.java

```diff
@@ -13,29 +13,177 @@
  */
 package io.trino.plugin.hive.avro;
 
+import io.airlift.log.Logger;
 import org.apache.avro.Schema;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.hive.serde2.SerDeUtils;
 import org.apache.hadoop.hive.serde2.avro.AvroSerDe;
 import org.apache.hadoop.hive.serde2.avro.AvroSerdeException;
 import org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils;
+import org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils.AvroTableProperties;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoUtils;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Properties;
 
+import static org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils.EXCEPTION_MESSAGE;
+import static org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils.SCHEMA_NONE;
+
 public class TrinoAvroSerDe
         extends AvroSerDe
 {
+    private static final Logger LOG = Logger.get(TrinoAvroSerDe.class);
+
     @Override
     public Schema determineSchemaOrReturnErrorSchema(Configuration conf, Properties props)
     {
         // AvroSerDe does not propagate initialization exceptions. Instead, it stores just an exception's message in
         // this.configErrors (see https://issues.apache.org/jira/browse/HIVE-7868). In Trino, such behavior is not
         // at all useful, as silenced exception usually carries important information which may be otherwise unavailable.
         try {
-            return AvroSerdeUtils.determineSchemaOrThrowException(conf, props);
+            return determineSchemaOrThrowException(conf, props);
         }
         catch (IOException | AvroSerdeException e) {
             throw new RuntimeException(e);
         }
     }
+
+    /**
+     * Copied from {@link AvroSerdeUtils#determineSchemaOrThrowException(Configuration, Properties)
+     * AvroSerdeUtils} with no changes besides formatting.
```

Comment:
I've simplified it by removing the extra details, but I think the basic "copied from" message is fitting as javadoc, because it refers the user to the original (and makes a link that's easy to follow in IntelliJ).

Maybe a better option would be to put it in a separate file with one comment at the top to make it clear that the code isn't original to Trino?

(But also see my other comment.)